src/transformStateWithClones.js

```diff
@@ -7,7 +7,23 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const transformHistory = [];
+  let transformStep = { ...state };
+
+  for (const element of actions) {
+    if (element.type === 'addProperties') {
+      Object.assign(transformStep, element.extraData);
```

Comment:
You could use destructuring here, since you can reassing variable `transformStep`

```suggestion
      transformStep = { 
        ...transformStep,
        ...action.extraData,
      };
```

Looks more concise)